roachtest: django failed

roachtest.django [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6315055&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6315055&tab=artifacts#/django) on release-22.1 @ [5e2f4ff465491982e869cfb113fe9d73a087b081](https://github.com/cockroachdb/cockroach/commits/5e2f4ff465491982e869cfb113fe9d73a087b081):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/django/run_1
	orm_helpers.go:193,orm_helpers.go:119,django.go:235,django.go:246,test_runner.go:883: 
		Tests run on Cockroach v22.1.6-112-g5e2f4ff465
		Tests run against django cockroach-4.1.x
		9939 Total Tests Run
		9938 tests passed
		1 test failed
		642 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: select_for_update.tests.SelectForUpdateTests.test_raw_lock_not_available (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist22_1) is available in the artifacts' django log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*django.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-19235

roachtest.django [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6347321&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6347321&tab=artifacts#/django) on release-22.1 @ [443dd822332bb6487c949efbd9a4b700693850d4](https://github.com/cockroachdb/cockroach/commits/443dd822332bb6487c949efbd9a4b700693850d4):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/django/run_1
	orm_helpers.go:193,orm_helpers.go:119,django.go:235,django.go:246,test_runner.go:883: 
		Tests run on Cockroach v22.1.6-125-g443dd82233
		Tests run against django cockroach-4.1.x
		9931 Total Tests Run
		9929 tests passed
		2 tests failed
		642 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: migrations.test_operations.OperationTests.test_rename_model_with_db_table_noop (unexpected)
		--- FAIL: unittest.loader._FailedTest.admin_docs.test_utils (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist22_1) is available in the artifacts' django log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*django.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.django [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6354756&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6354756&tab=artifacts#/django) on release-22.1 @ [a70cc0eed145522afbf1e9f0307fb5c554562050](https://github.com/cockroachdb/cockroach/commits/a70cc0eed145522afbf1e9f0307fb5c554562050):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/django/run_1
	orm_helpers.go:193,orm_helpers.go:119,django.go:235,django.go:246,test_runner.go:883: 
		Tests run on Cockroach v22.1.6-127-ga70cc0eed1
		Tests run against django cockroach-4.1.x
		9931 Total Tests Run
		9929 tests passed
		2 tests failed
		642 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: migrations.test_operations.OperationTests.test_rename_model_with_db_table_noop (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: unittest.loader._FailedTest.admin_docs.test_utils (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist22_1) is available in the artifacts' django log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #87368 roachtest: django failed [C-test-failure O-roachtest O-robot branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*django.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.django [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6369525&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6369525&tab=artifacts#/django) on release-22.1 @ [a70cc0eed145522afbf1e9f0307fb5c554562050](https://github.com/cockroachdb/cockroach/commits/a70cc0eed145522afbf1e9f0307fb5c554562050):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/django/run_1
	orm_helpers.go:193,orm_helpers.go:119,django.go:235,django.go:246,test_runner.go:883: 
		Tests run on Cockroach v22.1.6-127-ga70cc0eed1
		Tests run against django cockroach-4.1.x
		9931 Total Tests Run
		9929 tests passed
		2 tests failed
		642 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: unittest.loader._FailedTest.admin_docs.test_utils (unexpected)
		--- FAIL: migrations.test_operations.OperationTests.test_rename_model_with_db_table_noop (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist22_1) is available in the artifacts' django log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #87368 roachtest: django failed [C-test-failure O-roachtest O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*django.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.django [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6383269&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6383269&tab=artifacts#/django) on release-22.1 @ [ad4b21c4707741e4b53fdc77a1fe01b1a8ac6fa5](https://github.com/cockroachdb/cockroach/commits/ad4b21c4707741e4b53fdc77a1fe01b1a8ac6fa5):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/django/run_1
	orm_helpers.go:193,orm_helpers.go:119,django.go:235,django.go:246,test_runner.go:883: 
		Tests run on Cockroach v22.1.6-139-gad4b21c470
		Tests run against django cockroach-4.1.x
		9941 Total Tests Run
		9939 tests passed
		2 tests failed
		642 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: schema.tests.SchemaTests.test_autofield_to_o2o (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: schema.tests.SchemaTests.test_add_auto_field (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist22_1) is available in the artifacts' django log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #87368 roachtest: django failed [C-test-failure O-roachtest O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*django.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/blocklist_test.go
+++ b/pkg/cmd/roachtest/tests/blocklist_test.go
@@ -36,7 +36,7 @@ func TestBlocklists(t *testing.T) {
 		"hibernate":    hibernateBlockList20_2,
 		"pgjdbc":       pgjdbcBlockList,
 		"psycopg":      psycopgBlockList20_2,
-		"django":       djangoBlocklist20_2,
+		"django":       djangoBlocklist,
 		"sqlAlchemy":   sqlAlchemyBlocklist20_2,
 		"libpq":        libPQBlocklist20_2,
 		"gopg":         gopgBlockList20_2,

--- a/pkg/cmd/roachtest/tests/django.go
+++ b/pkg/cmd/roachtest/tests/django.go
@@ -28,7 +28,7 @@ var djangoReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)\.(?P<minor>\d+)(
 var djangoCockroachDBReleaseTagRegex = regexp.MustCompile(`^(?P<major>\d+)\.(?P<minor>\d+)$`)
 
 var djangoSupportedTag = "cockroach-4.1.x"
-var djangoCockroachDBSupportedTag = "4.1"
+var djangoCockroachDBSupportedTag = "4.1.*"
 
 func registerDjango(r registry.Registry) {
 	runDjango := func(
@@ -176,13 +176,7 @@ func registerDjango(r registry.Registry) {
 			t.Fatal(err)
 		}
 
-		blocklistName, expectedFailureList, ignoredlistName, ignoredlist := djangoBlocklists.getLists(version)
-		if expectedFailureList == nil {
-			t.Fatalf("No django blocklist defined for cockroach version %s", version)
-		}
-		if ignoredlist == nil {
-			t.Fatalf("No django ignorelist defined for cockroach version %s", version)
-		}
+		blocklistName, ignoredlistName := "djangoBlocklist", "djangoIgnoreList"
 		t.L().Printf("Running cockroach version %s, using blocklist %s, using ignoredlist %s",
 			version, blocklistName, ignoredlistName)
 
@@ -219,9 +213,9 @@ func registerDjango(r registry.Registry) {
 		t.Status("collating test results")
 
 		results := newORMTestsResults()
-		results.parsePythonUnitTestOutput(fullTestResults, expectedFailureList, ignoredlist)
+		results.parsePythonUnitTestOutput(fullTestResults, djangoBlocklist, djangoIgnoreList)
 		results.summarizeAll(
-			t, "django" /* ormName */, blocklistName, expectedFailureList, version, djangoSupportedTag,
+			t, "django" /* ormName */, blocklistName, djangoBlocklist, version, djangoSupportedTag,
 		)
 	}
 

--- a/pkg/cmd/roachtest/tests/django_blocklist.go
+++ b/pkg/cmd/roachtest/tests/django_blocklist.go
@@ -161,60 +161,10 @@ var enabledDjangoTests = []string{
 	"view_tests",
 }
 
-var djangoBlocklists = blocklistsForVersion{
-	{"v20.2", "djangoBlocklist20_2", djangoBlocklist20_2, "djangoIgnoreList20_2", djangoIgnoreList20_2},
-	{"v21.1", "djangoBlocklist21_1", djangoBlocklist21_1, "djangoIgnoreList21_1", djangoIgnoreList21_1},
-	{"v21.2", "djangoBlocklist21_2", djangoBlocklist21_2, "djangoIgnoreList21_2", djangoIgnoreList21_2},
-	{"v22.1", "djangoBlocklist22_1", djangoBlocklist22_1, "djangoIgnoreList22_1", djangoIgnoreList22_1},
-	{"v22.2", "djangoBlocklist22_2", djangoBlocklist22_2, "djangoIgnoreList22_2", djangoIgnoreList22_2},
-}
-
 // Maintain that this list is alphabetized.
-var djangoBlocklist22_2 = djangoBlocklist22_1
+var djangoBlocklist = blocklist{}
 
-var djangoBlocklist22_1 = djangoBlocklist21_2
-
-var djangoBlocklist21_2 = djangoBlocklist21_1
-
-var djangoBlocklist21_1 = djangoBlocklist20_2
-
-var djangoBlocklist20_2 = blocklist{}
-
-var djangoIgnoreList22_2 = djangoIgnoreList22_1
-
-var djangoIgnoreList22_1 = djangoIgnoreList21_2
-
-var djangoIgnoreList21_2 = blocklist{
-	"migrations.test_operations.OperationTests.test_alter_fk_non_fk":  "will be fixed in django-cockroachdb v3.2.2",
-	"schema.tests.SchemaTests.test_alter_field_db_collation":          "will be fixed in django-cockroachdb v3.2.2",
-	"schema.tests.SchemaTests.test_alter_field_type_and_db_collation": "will be fixed in django-cockroachdb v3.2.2",
-}
-
-var djangoIgnoreList21_1 = blocklist{
-	"schema.tests.SchemaTests.test_alter_field_db_collation":          "will be fixed in django-cockroachdb v3.2.2",
-	"schema.tests.SchemaTests.test_alter_field_type_and_db_collation": "will be fixed in django-cockroachdb v3.2.2",
-}
-
-var djangoIgnoreList20_2 = blocklist{
-	"expressions.tests.BasicExpressionsTests.test_boolean_expression_combined":   "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_attribute_name_not_python_keyword":   "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_digits_column_name_introspection":    "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_field_types":                         "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_managed_models":                      "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_number_field_types":                  "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_special_column_name_introspection":   "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_stealth_table_name_filter_option":    "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_table_name_introspection":            "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_table_option":                        "unknown",
-	"inspectdb.tests.InspectDBTestCase.test_unique_together_meta":                "unknown",
-	"introspection.tests.IntrospectionTests.test_get_constraints_indexes_orders": "unknown",
-	"introspection.tests.IntrospectionTests.test_get_table_description_types":    "unknown",
-	"schema.tests.SchemaTests.test_add_field_temp_default":                       "unknown",
-	"schema.tests.SchemaTests.test_alter":                                        "unknown",
-	"schema.tests.SchemaTests.test_alter_field_fk_keeps_index":                   "unknown",
-	"schema.tests.SchemaTests.test_alter_field_fk_to_o2o":                        "unknown",
-	"schema.tests.SchemaTests.test_alter_numeric_field_keep_null_status":         "unknown",
-	"schema.tests.SchemaTests.test_alter_smallint_pk_to_smallautofield_pk":       "unknown",
-	"schema.tests.SchemaTests.test_db_table":                                     "unknown",
-	"schema.tests.SchemaTests.test_foreign_key_index_long_names_regression":      "unknown",
+var djangoIgnoreList = blocklist{
+	"schema.tests.SchemaTests.test_add_auto_field":   "unneeded once django-cockroachdb 4.1.1 is released",
+	"schema.tests.SchemaTests.test_autofield_to_o2o": "unneeded once django-cockroachdb 4.1.1 is released",
 }

